src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsnonassociationsapi/jpa/NonAssociation.kt

```diff
@@ -18,7 +18,7 @@ import uk.gov.justice.digital.hmpps.hmppsnonassociationsapi.dto.NonAssociation a
 
 @Entity
 @EntityListeners(AuditingEntityListener::class)
-class NonAssociation(
+data class NonAssociation(
```

Comment:
Without this couldn't clone the object!

Reply:
I haven't looked at the PR yet but the choice of _**not**_ using data classes _for entities_ was deliberate.

IIRC docs mention to not use data classes for JPA entities, it was also mentioned to avoid doing so [in this article](https://jpa-buddy.com/blog/best-practices-and-common-pitfalls/) and when I was playing with some entities some time ago I was noticing some strange behaviour (but it was some time ago and I don't remember exactly sorry)

Are we sure this is right to do and everything will continue to work as expected (all the usual JPA machinery) even tho is a bit less convenient? (I appreciate you can't copy a entity but it can be updated if necessary, and if a field should be modified we can change to a `var` field, if it makes sense)

Reply:
oh yeah i forgot about this. the spring docs do definitely say you should not use immutable classes (i.e. data classes) with JPA because the proxy wizardry silently fails sometimes.

for the purpose of this PR, you really are mutating an existing record so there's no requirement to make a copy of it. you can just change it in place.

Reply:
yeah agreed, change the `firstPrisonerNumber`/`secondPrisonerNumber` fields to be mutable (`var` instead of `val`) should do and it's nice because they could change (as in this case of a merge event which update records' prisoner numbers)